consensus/src/dag/dag_handler.rs

```diff
@@ -47,13 +48,20 @@ impl NetworkHandler {
     async fn start(mut self) {
         while let Some(msg) = self.dag_rpc_rx.next().await {
             if let Err(e) = self.process_rpc(msg).await {
-                warn!(error = ?e, "error sending rpc response for request");
+                warn!(error = ?e, "error processing rpc");
             }
         }
     }
 
     async fn process_rpc(&mut self, rpc_request: IncomingDAGRequest) -> anyhow::Result<()> {
-        let dag_message: DAGMessage = TConsensusMsg::from_network_message(rpc_request.req)?;
+        let dag_message: DAGMessage = rpc_request.req.try_into()?;
+
+        let author = dag_message
+            .author()
+            .map_err(|_| anyhow::anyhow!("unexpected rpc message {:?}", dag_message))?;
+        if author != rpc_request.sender {
```

Comment:
we can still use ensure here

Reply:
Yes, but ideally, we would log this here, so I kept this an if statement.